Tribler/Main/Build/Win/tribler.nsi

```diff
@@ -1,358 +1,361 @@
+SectionEnd
+
+
+Section "Make Default For .tstream" SecDefaultTStream
+   ; Arno: Poor man's attempt to check if already registered
+   ReadRegStr $0 HKCR .tstream ""
+   ReadRegStr $1 HKCR "tstream\shell\open\command" ""
+   StrCpy $2 $1 -4
+   StrCmp $0 "" 0 +2
+      return
+   MessageBox MB_YESNO ".tstream already registered to $2. Overwrite?" IDYES +2 IDNO 0
+      Return
+   DetailPrint "Arno registering .tstream: $0 $1 $2"
+
+   ; Register
+   WriteRegStr HKCR .tstream "" tstream
+   WriteRegStr HKCR .tstream "Content Type" application/x-tribler-stream
+   WriteRegStr HKCR "MIME\Database\Content Type\application/x-tribler-stream" Extension .tstream
+   WriteRegStr HKCR tstream "" "TSTREAM File"
+   WriteRegBin HKCR tstream EditFlags 00000100
+   WriteRegStr HKCR "tstream\shell" "" open
+   WriteRegStr HKCR "tstream\shell\open\command" "" '"$INSTDIR\${PRODUCT}.exe" "%1"'
+   WriteRegStr HKCR "tstream\DefaultIcon" "" "$INSTDIR\Tribler\Main\vwxGUI\images\torrenticon.ico"
+SectionEnd
+
+Section "Make Default For magnet://" SecDefaultMagnet
+   WriteRegStr HKCR "magnet" "" "URL: Magnet Link Protocol"
+   WriteRegStr HKCR "magnet" "URL Protocol" ""
+   WriteRegStr HKCR "magnet\DefaultIcon" "" "$INSTDIR\Tribler\Main\vwxGUI\images\torrenticon.ico"
+   WriteRegStr HKCR "magnet\shell\open\command" "" '"$INSTDIR\${PRODUCT}.exe" "%1"'
+   WriteRegStr HKLM "SOFTWARE\Classes\magnet\shell\open\command" "" '"$INSTDIR\${PRODUCT}.exe" "%1"'
+SectionEnd
+
+;--------------------------------
+;Descriptions
+
+!insertmacro MUI_FUNCTION_DESCRIPTION_BEGIN
+!insertmacro MUI_DESCRIPTION_TEXT ${SecMain} $(DESC_SecMain)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecDesk} $(DESC_SecDesk)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecStart} $(DESC_SecStart)
+;!insertmacro MUI_DESCRIPTION_TEXT ${SecLang} $(DESC_SecLang)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecDefaultTorrent} $(DESC_SecDefaultTorrent)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecDefaultTStream} $(DESC_SecDefaultTStream)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecDefaultMagnet} $(DESC_SecDefaultMagnet)
+!insertmacro MUI_DESCRIPTION_TEXT ${SecDefaultPpsp} $(DESC_SecDefaultPpsp)
+!insertmacro MUI_FUNCTION_DESCRIPTION_END
+
+;--------------------------------
+;Uninstaller Section
+
+Section "Uninstall"
+
+ RMDir /r "$INSTDIR"
+
+ Delete "$DESKTOP\${PRODUCT}.lnk"
+
+ SetShellVarContext all
+ RMDir "$SMPROGRAMS\${PRODUCT}"
+ RMDir /r "$SMPROGRAMS\${PRODUCT}"
+
+ DeleteRegKey HKEY_LOCAL_MACHINE "SOFTWARE\${PRODUCT}"
+ DeleteRegKey HKEY_LOCAL_MACHINE "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}"
+
+ ; Remove an application from the firewall exception list
+ SimpleFC::RemoveApplication "$INSTDIR\${PRODUCT}.exe"
+ ; Pop $0 ; return error(1)/success(0)
+
+SectionEnd
+
+;--------------------------------
+;Macros and Functions Section
+
+
+!macro Init thing
+uac_tryagain:
+!insertmacro UAC_RunElevated
+
+; Currently this throws error 1223 on newer windows systems, but this can be ignored.
```

Comment:
If you move and modify a file in the same commit, git can't know that it actually was the same file. github seems to be doing something "smart" but on the history it will still look bad.